airbyte-integrations/connectors/source-google-search-console/source_google_search_console/streams.py

```diff
@@ -295,3 +295,52 @@ class SearchAnalyticsByQuery(SearchAnalytics):
 
 class SearchAnalyticsAllFields(SearchAnalytics):
     dimensions = ["date", "country", "device", "page", "query"]
+
+
+class SearchAnalyticsByCustomDimensions(SearchAnalytics):
+    def __init__(self, dimensions: List[str], *args, **kwargs):
+        super(SearchAnalyticsByCustomDimensions, self).__init__(*args, **kwargs)
+        self.dimensions = dimensions
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return super().cursor_field if "date" in self.dimensions else []
```

Comment:
If the dimensions list doesn't include a date we will do. a full sync?

Reply:
yes, as airbyte sends nulls as checkpoints when the `date` is not present, which breaks the next run